### PR TITLE
Reserve Bounties

### LINKED ISSUE #2037: BUILD - Reserve Bounties
### User Story

[comment]: # (As a <user type>, I want to <task> so that <goal>.)
As a funder, I want to fund and send a bounty directly to a certain group of individuals (or perhaps, just one individual).

### Why Is this Needed
This is needed so that select group of individuals can work on a project.

[comment]: # (Describe the problem and why this task is needed. Provide description of the current state, what you would like to happen, and what actually happen)
*Summary*: Right now, I sometimes get unnecessary noise on bounties that I would rather send directly to a smaller group of people that I know from prior work on the platform. 

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

### Current Behavior
[comment]: # (Describe what actually happened.)
Post a bounty on the main issue explorer, make it approval-only, and approve the worker of my choosing. If I don't get to approve within 24 hours, perhaps deal with auto-approvals of people I didn't want to get involved. 

### Definition of Done
[comment]: # (Any other information that would be useful, bullets are helpful.)

**Funder Form**
- [ ] On the funder form under Permissions Field add the following: Reserved for ___________________.
- [ ] User should be able to type in a username in the input field

**Issue Details**
- [ ] The Issue Details should display that this issue is Reserved for X Contributor with the contributor's icon, linking to their profile.
- [ ] Only the contributor(s) that have been selected to work this issue can click start work.
- [ ] Group information and line the information up like this (Will Tip Extra)
screen shot 2018-08-27 at 7 43 16 am

**Email**
- [ ] User(s) should be notified that an issue has been reserved for them to work on.

### Reviewed by 
[comment]: # (Any other information that would be useful, content, screenshots, etc.)

@owocki @PixelantDesign @thelostone-mc Thoughts?

### PR DESCRIPTION
**Pull request by mul1sh**
##### Description

This PR allows funders to reserve bounties for particular users for 72 hours.

##### Checklist

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)

web

##### Testing

Yes I have tested this PR, and it will not break anything when merged into master

##### Refers/Fixes

Fixes #2037

### CODE REVIEW

**Comment by mul1sh** [comment]
This PR of a duplicate one I closed a few days ago, https://github.com/gitcoinco/web/pull/2201.

I closed the old PR out of frustration because of the time it was taking to review the changes 😞 , but I have since learned that, that was wrong of me, I should be  more patient with the review process and not jump the gun needlessly.

So anyway i'll gladly work with the review team till all issues are sorted and this PR is merged and the ticket closed 🙂 , thanks

**Comment by vs77bb** [comment]
@mul1sh In talking to @mbeacom, there was some feedback on the last PR here which was unresolved regarding client side filtering of bounties. Are you able to access that feedback still? That would be a good starting place to continue work on this issue.

**Comment by mul1sh** [comment]
@vs77bb @mbeacom  I believe the backend filtering was for another PR #2279  and i've since moved the filtering to the backend as required.

For this PR though i don't think there is any backend filtering unless i'm wrong 🙂

**Comment by mbeacom** [comment]
:star: PR built and available in a preview environment **gitcoinco-web-pr-2275** [here](http://web.jx-gitcoinco-web-pr-2275.ci.gitcoin.co)

File: app/assets/v2/js/pages/new_bounty.js [hunk 1]
```diff
@@ -10,6 +10,8 @@ var new_bounty = {
   last_sync: new Date()
 };
 
+var bounty_reserved_for_user = null;
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/new_bounty.js]: update it to `let` ?

File: app/retail/templates/emails/reserved_issue.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+     This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Affero General Public License for more details.
+     You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+ {% endcomment %}
+{% load i18n humanize %}
+{% load static %}
+ {% block content %}
```
Comment (thelostone-mc) [inline app/retail/templates/emails/reserved_issue.html]: indent fix needed

File: app/retail/templates/emails/reserved_issue.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ <h1>{% trans "Reserved Issue" %}</h1>
+<div style="text-align: left! important">
```
Comment (thelostone-mc) [inline app/retail/templates/emails/reserved_issue.html]: could we indent this div container by 1 ?

**Comment by PixelantDesign** [comment]
@mul1sh are you still interested in working on this? We haven't heard from you we'll need to pass this to another contributor.

**Comment by mul1sh** [comment]
@PixelantDesign oops sorry for that I had forgotten to finish this off but  i'll sort it out tomorrow

**Comment by pinkiebell** [comment]
@mul1sh
I already worked on this PR a bit, here is my branch for reference: 
https://github.com/pinkiebell/web/tree/reservebounties

You can check it out and incorporate my changes if you want 👍

**Comment by mul1sh** [comment]
@pinkiebell awesome thanks you so much for the feedback and the PR, i'm sorting it now 😃

**Comment by PixelantDesign** [comment]
Thanks @pinkiebell @mul1sh we'd love to get this out next week!

**Comment by PixelantDesign** [comment]
@pinkiebell @mul1sh any updates here? We'll need to bring this in house if I don't hear back in the next 48 hrs. Thanks!

**Comment by mul1sh** [comment]
btw @pinkiebell i saw you made a PR #2950 with my old changes, so to be fair I'll update this one too and then @PixelantDesign can pick the most relevant

**Comment by PixelantDesign** [comment]
Thanks! @mul1sh

**Comment by mul1sh** [comment]
closed this PR in favour of #2950 which a follow-up to this one

**Pull request closed without merging**